### ISSUE TITLE
Modal dialog title and heading are announced multiple times

### ISSUE THREAD

**Issue by gkhadK** [vmware/clarity#4595]
**Clarity Version:** 3.1.1
**WCAG 2.1 Guideline:** 2.4.6 Headings and Labels

**Test Environment:**
- Test URL: https://clarity.design/documentation/modals
- OS: Windows 10
- User Agents/Assistive Technologies: Firefox 76.0
- Screen reader: NVDA 19.3.1

**Steps to reproduce:**
1. Open the test URL.
2. Move focus onto to the first Show Modal button present under Angular modals.
3. Activate the button by pressing the enter or space key and observe the spoken feedback for title announcement.

**Expected result:**
When the dialog is opened, title should be announced only once, then rest of content should be announced skipping heading announcement to avoid redundancy.

**Actual result:**
Title is announced 4 times in the following way:
1. I have nice title, dialog.
2. I have nice title, heading level 3.
3. I have nice title, heading level 3.
4. I have nice title.

**Reporter Email:** gkhadke@vmware.com

**Comment by immihaylova** [comment]
SRM UI product has an issue filed by AccessLevel with Severity 1 (lowest) describing this behavior.

**Comment by immihaylova** [comment]
I found smth in the AccessLevel report that might help with the investigation of this issue:

> When the modal opens, the focus is set on the header. But because the modal structure is already using aria-describedby referencing the id of the header, screen reader is conveying the title three times.

**Comment by gnomeontherun** [comment]
Hi there, this is an automated message.

In an effort to dedupe and track accessibility reports consistently, this issue is being closed and tracked internally with our Accessibility team. It will still be triaged and addressed as part of ongoing work.
